src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java

```diff
@@ -351,7 +354,7 @@ public Object convert(Class type, Object value) {
                 }
             }
 
-            return result;
+            return (Object) result;
```

Comment:
Was this actually required?

Reply:
@nrmancuso Typecasting was needed to resolve the incompatibility issue between expression and return type .

Reply:
>Can you show what the failure looks like if we don’t have the cast here?

Please show exact failure related to not casting the return value here.

Reply:
@nrmancuso yeah you were right . It was actually not needed. I did this change due to Circle CI failure. But now after removing typecasting all the checks are passing